### DISCUSSION TITLE
Rule to make method argument required (remove the default value)

### DISCUSSION THREAD

**Discussion by brandonkelly** [rectorphp/rector#8240] [Ideas]
It would be great if there was a Rector rule that removes the default value from a method argument (including subclasses), making it required.

```diff
- public function foo(?string $foo = null) {
+ public function foo(?string $foo) {
      // ...
  }
```

It should also modify the type declaration for types that were made nullable thanks to the default value (this is how you could have a nullable type prior to PHP 7.1):

```diff
- public function bar(string $foo = null) {
+ public function bar(?string $foo) {
      // ...
  }

- public function baz(string|int $foo = null) {
+ public function baz(string|int|null $foo) {
      // ...
  }
```

We could use this rule in [craftcms/rector](https://github.com/craftcms/rector) to help third party plugins clean up their code, when they’re overriding a method that used to define a default argument in the base class/interface, but doesn’t any longer.

**Comment by TomasVotruba** [comment]
I'll look into this 👍
